Sources/CvPlayer.cpp

```diff
@@ -5151,10 +5122,9 @@ void CvPlayer::doTurn()
 /* REVOLUTION_MOD                          END                                                  */
 /************************************************************************************************/
 
-	CvUnit* pLoopUnit;
-
-	for(pLoopUnit = firstUnit(&iLoop); pLoopUnit != NULL; pLoopUnit = nextUnit(&iLoop))
+	for (CvPlayer::unit_iterator unitItr = beginUnits(); unitItr != endUnits(); ++unitItr)
 	{
+		CvUnit* pLoopUnit = *unitItr;
```

Comment:
ah yes, I tried the second way andit didn't work.

Reply:
Yeah cos the underlying container stores pointers so the iterator needs de-referencing twice. I'm going to make the iterators return references regardless though, one benefit is it will remove ambiguity that sometimes these objects could just randomly be null. From now on if they are null it is an error in our code we need to fix.